app/controllers/csvs_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CsvsController < ApplicationController
+
+  def index
+    respond_to do |format|
+      @csv_exports = CsvExport.where(user_id: current_user.id)
+      format.html
+      format.json { render json: @csv_exports }
+    end
+  end
+  
+  def show
+    respond_to do |format|
+      @csv_export = CsvExport.find(params[:id])
+      format.html
+      format.json { render json: @csv_export }
+      format.csv { download(params) }
+    end
+  end
+
+  def create
+    csv_export = CsvExport.create!(user: current_user, content: params[:content], filters: params.to_json)
+    csv_export.pending!
+    CsvExportJob.perform_later(csv_export.id)
+    redirect_to action: "show", id: csv_export.id
+  end
+
+  private
+
+  def download(params)
+    csv_export = CsvExport.find(params[:id])
+    check_file(csv_export)
+    if csv_export.status? :ready
+      send_file csv_export.full_filename, type: :csv, filename: csv_export.filename
+      csv_export.downloaded!
+      Rails.logger.info("#{current_user.name_and_email} just downloaded #{csv_export.filename})")
+    else
+      redirect_to action: 'show', id: params[:id]
+    end
+  end
+
+  def check_file(csv_export)
+    if csv_export.status? :ready
```

Comment:
same condition as above ... this would mark it at deleted and then try to download ... missing testcase ?